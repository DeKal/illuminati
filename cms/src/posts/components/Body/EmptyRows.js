import React from 'react'
import PropTypes from 'prop-types'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import { DEFAULT_LINE_HEIGHT } from 'core/const/posts'

const EmptyRows = ({ page, rowsPerPage, postsLength }) => {
  const emptyRows =
    rowsPerPage - Math.min(rowsPerPage, postsLength - page * rowsPerPage)
  return emptyRows ? (
    <TableRow style={{ height: DEFAULT_LINE_HEIGHT * emptyRows }}>
      <TableCell colSpan={6} />
    </TableRow>
  ) : null
}

export default EmptyRows

EmptyRows.propTypes = {
  page: PropTypes.number,
  rowsPerPage: PropTypes.number,
  postsLength: PropTypes.number
}
