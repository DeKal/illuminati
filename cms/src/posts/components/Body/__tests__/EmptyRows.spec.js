import * as React from 'react'
import { shallow } from 'enzyme'
import EmptyRows from '../EmptyRows'

describe('EmptyRows', () => {
  it('should render EmptyRows when postLength < rows per page', function() {
    const wrap = shallow(
      <EmptyRows page={0} rowsPerPage={20} postsLength={15} />
    )
    expect(wrap).toMatchSnapshot()
  })
  it('should render EmptyRows when postLength > rows per page', function() {
    const wrap = shallow(
      <EmptyRows page={0} rowsPerPage={20} postsLength={21} />
    )
    expect(wrap).toMatchSnapshot()
  })
})
