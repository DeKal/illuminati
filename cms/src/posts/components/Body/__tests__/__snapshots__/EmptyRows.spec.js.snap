// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyRows should render EmptyRows when postLength < rows per page 1`] = `
<WithStyles(ForwardRef(TableRow))
  style={
    Object {
      "height": 265,
    }
  }
>
  <WithStyles(ForwardRef(TableCell))
    colSpan={6}
  />
</WithStyles(ForwardRef(TableRow))>
`;

exports[`EmptyRows should render EmptyRows when postLength > rows per page 1`] = `""`;
